File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {
    // define expected cooking time in minutes
    private static final int EXPECTED_OVEN_TIME_IN_MINUTES = 40;
    // define expected prep time per layer 
    private static final int PREPARATION_MINUTES_PER_LAYER = 2;

    // define expected minutes in oven for normal pizza
    public int expectedMinutesInOven() {
        return EXPECTED_OVEN_TIME_IN_MINUTES;
    }
    // given minutes in oven determine remaining minutes
    public int remainingMinutesInOven(int actualMinutesInOven) {
        return EXPECTED_OVEN_TIME_IN_MINUTES - actualMinutesInOven;
    }

    // given number of layers, determine prep time
    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * PREPARATION_MINUTES_PER_LAYER;
    }

    // given layer prep time and minutes in the oven, determine total time to make the pizza
    public int totalTimeInMinutes(int numberOfLayers, int minutesInOven) {
        return preparationTimeInMinutes(numberOfLayers) + minutesInOven;
    }
}

File: solutions/elixir/lasagna/1/lib/lasagna.ex
defmodule Lasagna do
  # define the expected time in the oven in minutes for the lasagne
  def expected_minutes_in_oven do
    40
  end

  # calculate the remaining time in the oven in minutes for the lasagne
  def remaining_minutes_in_oven(minutes) do
    expected_minutes_in_oven() - minutes
  end

  # calculate the prep time in minutes for the lasagne, given the number of layers
  def preparation_time_in_minutes(layers) do
    2 * layers
  end

  # calculate the total working time in minutes for the lasagne, given layers and time in the oven
  def total_time_in_minutes(layers, minutes) do
    preparation_time_in_minutes(layers) + minutes
  end

  # return "Ding!" when an alarm is requested
  def alarm do
    "Ding!"
  end

end
